src/darwin/Framework/CHIP/MTRDeviceController_Internal.h

```diff
@@ -243,6 +245,23 @@ NS_ASSUME_NONNULL_BEGIN
                             queue:(dispatch_queue_t)queue
                        completion:(void (^)(NSURL * _Nullable url, NSError * _Nullable error))completion;
 
+/**
+ * Get the access grants that apply for the given cluster path.
+ */
+- (NSArray<MTRAccessGrant *> *)accessGrantsForClusterPath:(MTRClusterPath *)clusterPath;
+
+/**
+ * Get the privilege level needed to read the given attribute.  There's no
+ * endpoint provided because the expectation is that this information is the
```

Comment:
Sounds like this is an assumption of the SDK? Maybe this should be validated when endpoints are activated, and fail addition of the endpoint if it contains any attribute definitions that disagree with existing ones.

Reply:
> Sounds like this is an assumption of the SDK

Yes, since the SDK assumes that clusters have a spec and the spec defines access levels.

> Maybe this should be validated when endpoints are activated

Good idea.  I will do that as a followup.

Reply:
https://github.com/project-chip/connectedhomeip/issues/31887